### DISCUSSION TITLE
Is there any way to print the complete file path using bpftrace?

### DISCUSSION THREAD

**Discussion by carbonvibes** [bpftrace/bpftrace#3109] [Q&A] [answered]
If my file path is greater than a certain limit, it is getting truncated, is there any way to obtain the full file path ??

```
/home/carbon/.local/share/tracker/data/tracker-store.ontology.j..
/sys/fs/cgroup/user.slice/user-1001.slice/user@1001.service/app..
```

I'm using openat to print the file location with create flag
```
if (args->flags & O_CREAT) {
        printf("%s\n",str(args->filename));
    }
```

**Comment by danobi** [comment, marked as answer]
Hi, you can try increasing the max string length: https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#max_strlen

**Reply by carbonvibes** [reply]
Thanks a lot!
